File: test-suite/src/test/java/io/micronaut/docs/jdbc/multiple/MultipleDataSourcesTest.java
package io.micronaut.docs.jdbc.multiple;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import java.sql.SQLException;

import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(transactional = false)
class MultipleDataSourcesTest {

    @Inject
    InventoryService inventoryService;

    @Test
    void testQualifiedDataSourcesAreInjected() throws SQLException {
        assertTrue(inventoryService.defaultUrl().contains("mem:default"));
        assertTrue(inventoryService.warehouseUrl().contains("mem:warehouse"));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/jdbc/multiple/MultipleDataSourcesTest.kt
package io.micronaut.docs.jdbc.multiple

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

@MicronautTest(transactional = false)
class MultipleDataSourcesTest {

    @Inject
    lateinit var inventoryService: InventoryService

    @Test
    fun testQualifiedDataSourcesAreInjected() {
        assertTrue(inventoryService.defaultUrl().contains("mem:default"))
        assertTrue(inventoryService.warehouseUrl().contains("mem:warehouse"))
    }
}
